server/websocket_connector.go

```diff
@@ -19,35 +19,35 @@ var webSocketUpgrader = websocket.Upgrader{
 }
 
 type WSHandler struct {
-	Router        PubSubSource
-	MessageSink   MessageSink
+	Router        Router
 	prefix        string
 	messageStore  store.MessageStore
 	accessManager auth.AccessManager
 }
 
-func NewWSHandler(prefix string) *WSHandler {
-	return &WSHandler{prefix: prefix}
-}
-
-func (handle *WSHandler) GetPrefix() string {
-	return handle.prefix
-}
-
-func (handle *WSHandler) SetMessageEntry(messageSink MessageSink) {
-	handle.MessageSink = messageSink
-}
+func NewWSHandler(
+	router Router,
+	prefix string) (*WSHandler, error) {
```

Comment:
No big deal, but I'm not a fan of line breaks in function arguments, if not absolutely needed.